dev/src/write-batch.ts

```diff
@@ -519,44 +521,48 @@ export class WriteBatch {
 
     this._committed = true;
 
-    return this._firestore.request('commit', request, tag).then(resp => {
-      const writeResults: WriteResult[] = [];
-
-      if (request.writes!.length > 0) {
-        assert(
-            resp.writeResults instanceof Array &&
-                request.writes!.length === resp.writeResults.length,
-            `Expected one write result per operation, but got ${
-                resp.writeResults.length} results for ${
-                request.writes!.length} operations.`);
-
-        const commitTime = Timestamp.fromProto(resp.commitTime);
-
-        let offset = 0;
-
-        for (let i = 0; i < this._writes.length; ++i) {
-          const writeRequest = this._writes[i];
-
-          // Don't return two write results for a write that contains a
-          // transform, as the fact that we have to split one write operation
-          // into two distinct write requests is an implementation detail.
-          if (writeRequest.write && writeRequest.transform) {
-            // The document transform is always sent last and produces the
-            // latest update time.
-            ++offset;
+    return this._firestore
+        .request<api.CommitResponse>(
+            'commit', request, tag, /* allowRetries= */ false)
+        .then(resp => {
+          const writeResults: WriteResult[] = [];
+
+          if (request.writes!.length > 0) {
+            assert(
+                Array.isArray(resp.writeResults) &&
```

Comment:
"resp.writeResults instanceof Array" to "Array.isArray(resp.writeResults)" is the only change in this block